src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -492,12 +492,21 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
-        return sprintf('alter table %s add %s %s%s(%s)',
+        $columns = collect($command->columns)->map(function ($column) {
+            // Check if the column contains a functional expression
+            if (preg_match('/\(.+\)/', $column)) {
+                return $column; // Return as-is if it's a functional expression
+            }
+            return $this->wrap($column); // Wrap normal columns
```

Comment:
Could be shortened to:
```suggestion
            // Check if the column contains a functional expression
            // and return as-is, otherwise wrap
            return preg_match('/\(.+\)/', $column) ? $column : $this->wrap($column);
```

Reply:
Maybe, you even want to put the `preg_match()` call into a separate method or the RegExp into a class constant so that the method/constant name speaks for itself